src/django_components/app_settings.py

```diff
@@ -93,95 +114,134 @@ class ContextBehavior(str, Enum):
     """
 
 
-class AppSettings:
+# This is the source of truth for the settings that are available. If the documentation
+# or the defaults do NOT match this, they should be updated.
+class ComponentsSettings(NamedTuple):
+    autodiscover: Optional[bool] = None
+    dirs: Optional[Sequence[Union[str, PathLike, Tuple[str, str], Tuple[str, PathLike]]]] = None
+    app_dirs: Optional[Sequence[str]] = None
+    context_behavior: Optional[ContextBehaviorType] = None
+    dynamic_component_name: Optional[str] = None
+    libraries: Optional[List[str]] = None
+    multiline_tags: Optional[bool] = None
+    reload_on_template_change: Optional[bool] = None
+    static_files_allowed: Optional[List[Union[str, re.Pattern]]] = None
+    forbidden_static_files: Optional[List[Union[str, re.Pattern]]] = None
+    tag_formatter: Optional[Union["TagFormatterABC", str]] = None
+    template_cache_size: Optional[int] = None
+
+
+# NOTE: Some defaults depend on the Django settings, which may not yet be
+# initialized at the time that these settings are generated. For such cases
+# we define the defaults as a factory function, and use the `Dynamic` class to
+# mark such fields.
+@dataclass(frozen=True)
+class Dynamic(Generic[T]):
+    getter: Callable[[], T]
+
+
+# This is the source of truth for the settings defaults. If the documentation
+# does NOT match it, the documentation should be updated.
+#
+# NOTE: Because we need to access Django settings to generate default dirs
+#       for `COMPONENTS.dirs`, we do it lazily.
+# NOTE 2: We show the defaults in the documentation, together with the comments
+#        (except for the `Dynamic` instances and comments like `type: ignore`).
+#        So `fmt: off` turns off Black formatting and `--snippet:defaults--` allows
+#        us to extract the snippet from the file.
+#
+# fmt: off
+# --snippet:defaults--
```

Comment:
In the docs, I embedded this snippet of default settings, so that it's kept in sync with the actual default values. How I did it was that at docs generation, this file is read, and the section between `--snippet:defaults--` and `--endsnippet:defaults--` is taken (and slightly modified).

Here's how it will look in the docs

<img width="1386" alt="Screenshot 2024-11-03 at 10 20 46" src="https://github.com/user-attachments/assets/40490143-106e-4269-8229-d88208c567b2">

Reply:
Really nice! :)